File: scala-source/com/barrybecker4/puzzle/tantrix/ui/TantrixViewer.scala
/*
 * // Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
 */
package com.barrybecker4.puzzle.tantrix.ui

import java.awt._

import com.barrybecker4.puzzle.common.ui.PuzzleViewer
import com.barrybecker4.puzzle.tantrix.model.{TantrixBoard, TilePlacement}
import com.barrybecker4.puzzle.tantrix.ui.rendering.TantrixBoardRenderer

/**
  * Draws the current best solution to the puzzle in a panel.
  * The view in the model-view-controller pattern.
  * @author Barry Becker
  */
final class TantrixViewer() extends PuzzleViewer[TantrixBoard, TilePlacement] {

  private val renderer: TantrixBoardRenderer = new TantrixBoardRenderer

  def getBoard: TantrixBoard = board

  /** This renders the current state of the PuzzlePanel to the screen.
    * This method is part of the component interface.
    */
  override protected def paintComponent(g: Graphics): Unit = {
    super.paintComponent(g)
    renderer.render(g, board, getWidth, getHeight)
  }

  override def refresh(board: TantrixBoard, numTries: Long): Unit = {
    status = createStatusMessage(numTries)
    simpleRefresh(board, numTries)
  }

  override def makeSound(): Unit = {
    val note = Math.min(127, 20 + getBoard.unplacedTiles.size * 12)
    musicMaker.playNote(note * 20, 20, 640)
  }
}

File: source/com/barrybecker4/puzzle/tantrix1/ui/TantrixViewer.java
// Copyright by Barry G. Becker, 2012. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.tantrix1.ui;

import com.barrybecker4.puzzle.common1.ui.PuzzleViewer;
import com.barrybecker4.puzzle.tantrix1.model.TantrixBoard;
import com.barrybecker4.puzzle.tantrix1.model.TilePlacement;
import com.barrybecker4.puzzle.tantrix1.ui.rendering.TantrixBoardRenderer;

import java.awt.*;

/**
 * Draws the current best solution to the puzzle in a panel.
 * The view in the model-view-controller pattern.
 *
 * @author Barry Becker
 */
public final class TantrixViewer extends PuzzleViewer<TantrixBoard, TilePlacement> {

    private TantrixBoardRenderer renderer_;

    /**
     * Constructor.
     */
    public TantrixViewer() {
        renderer_ = new TantrixBoardRenderer();
    }

    public TantrixBoard getBoard() {
        return board_;
    }

    /**
     *  This renders the current state of the PuzzlePanel to the screen.
     *  This method is part of the component interface.
     */
    @Override
    protected void paintComponent( Graphics g ) {

        super.paintComponent(g);
        renderer_.render(g, board_, getWidth(), getHeight());
    }


    @Override
    public void refresh(TantrixBoard board, long numTries) {
        //int rate = board_ == null ? 1 : board_.getNumTiles()-2;
        //System.out.println("rate=" + rate + " numTries="+ numTries+" numTries % rate=" + numTries % rate);
        //if (numTries % rate == 0) {
        status_ = createStatusMessage(numTries);
        simpleRefresh(board, numTries);
        //ThreadUtil.sleep(100);
        //}
    }

    public void makeSound() {
        int note = Math.min(127, 20 + getBoard().getUnplacedTiles().size() * 12);
        musicMaker_.playNote(note * 20, 20, 640);
    }
}

